File: movemork/Features/Welcome/WelcomeProofPreviewCard.swift
//
//  WelcomeProofPreviewCard.swift
//  movemork
//
//  Welcome-only saved proof preview — simpler than in-app ProofArtifactCard.
//

import SwiftUI

struct WelcomeProofPreviewCard: View {
    var imageName: String = "welcome-kitchen-main"
    var photoHeight: CGFloat = 160
    var cornerRadius: CGFloat = 20
    var tagsVisible: Bool = true

    private let issueTags = ["Already there", "Chipped paint"]

    @Environment(\.accessibilityReduceMotion) private var reduceMotion

    var body: some View {
        VStack(spacing: 0) {
            headerRow
            photoPane
            receiptFooter
        }
        .background(MoveMarkTheme.Colors.evidenceCard)
        .overlay(
            RoundedRectangle(cornerRadius: cornerRadius, style: .continuous)
                .stroke(MoveMarkTheme.Colors.cardStroke, lineWidth: 1)
        )
        .clipShape(RoundedRectangle(cornerRadius: cornerRadius, style: .continuous))
    }

    private var headerRow: some View {
        HStack(alignment: .center, spacing: 8) {
            Text("MOVE-IN PROOF")
                .font(.system(size: 10, weight: .semibold))
                .tracking(0.7)
                .foregroundStyle(MoveMarkTheme.Colors.textMuted)
                .lineLimit(1)
            Spacer(minLength: 8)
            Text("SAVED")
                .font(.system(size: 11, weight: .semibold))
                .foregroundStyle(MoveMarkTheme.Colors.primary.opacity(0.95))
                .padding(.horizontal, 9)
                .padding(.vertical, 5)
                .background(
                    Capsule()
                        .fill(MoveMarkTheme.Colors.primary.opacity(0.16))
                )
                .overlay(
                    Capsule()
                        .stroke(MoveMarkTheme.Colors.primary.opacity(0.48), lineWidth: 0.85)
                )
        }
        .padding(.horizontal, 14)
        .padding(.top, 12)
        .padding(.bottom, 10)
    }

    private var photoPane: some View {
        ZStack(alignment: .bottomLeading) {
            ZStack {
                Image(imageName)
                    .resizable()
                    .scaledToFill()
                    .frame(maxWidth: .infinity, maxHeight: .infinity)
                    .clipped()

                VStack {
                    Spacer(minLength: 0)
                    LinearGradient(
                        colors: [.clear, Color.black.opacity(0.42)],
                        startPoint: .top,
                        endPoint: .bottom
                    )
                    .frame(height: 56)
                }
            }
            .clipShape(RoundedRectangle(cornerRadius: 14, style: .continuous))

            if tagsVisible {
                HStack(spacing: 6) {
                    ForEach(Array(issueTags.enumerated()), id: \.offset) { index, tag in
                        Text(tag)
                            .font(.system(size: 10, weight: .semibold))
                            .foregroundStyle(Color.white.opacity(0.95))
                            .lineLimit(1)
                            .minimumScaleFactor(0.85)
                            .padding(.horizontal, 8)
                            .padding(.vertical, 4)
                            .background(
                                Color.black.opacity(0.72),
                                in: RoundedRectangle(cornerRadius: 8, style: .continuous)
                            )
                            .scaleEffect(reduceMotion ? 1 : (tagsVisible ? 1 : 0.92))
                            .animation(
                                reduceMotion ? nil : .easeOut(duration: 0.34).delay(Double(index) * 0.04),
                                value: tagsVisible
                            )
                    }
                }
                .padding(.leading, 12)
                .padding(.bottom, 12)
                .padding(.trailing, 8)
            }
        }
        .frame(maxWidth: .infinity)
        .frame(height: photoHeight)
        .padding(.horizontal, 14)
    }

    private var receiptFooter: some View {
        VStack(alignment: .leading, spacing: 4) {
            Rectangle()
                .fill(MoveMarkTheme.Colors.cardStroke.opacity(0.65))
                .frame(height: 1)
                .padding(.top, 12)

            HStack(alignment: .center, spacing: 6) {
                Image(systemName: "checkmark.circle.fill")
                    .font(.system(size: 14, weight: .semibold))
                    .foregroundStyle(MoveMarkTheme.Colors.primary.opacity(0.95))

                Text("Kitchen documented")
                    .font(.system(size: 15, weight: .semibold))
                    .foregroundStyle(MoveMarkTheme.Colors.textPrimary)
                    .lineLimit(1)
            }
            .padding(.top, 10)

            Text("12 photos · Verified Apr 14")
                .font(.system(size: 12, weight: .medium))
                .foregroundStyle(MoveMarkTheme.Colors.textMuted)
        }
        .frame(maxWidth: .infinity, alignment: .leading)
        .padding(.horizontal, 14)
        .padding(.bottom, 12)
    }
}

File: movemark-android/app/src/main/java/com/surensureshkumar/movemark/features/welcome/WelcomeProofPreviewCard.kt
package com.surensureshkumar.movemark.features.welcome

import androidx.compose.animation.core.animateFloatAsState
import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.CheckCircle
import androidx.compose.material3.Icon
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.scale
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.surensureshkumar.movemark.R
import com.surensureshkumar.movemark.core.design.MMColors
import com.surensureshkumar.movemark.core.design.MMMotion

private val welcomeIssueTags = listOf("Already there", "Chipped paint")

/**
 * Welcome-only saved proof preview — simpler than in-app [MMProofArtifactCard].
 */
@Composable
fun WelcomeProofPreviewCard(
    tagsVisible: Boolean,
    reduceMotion: Boolean,
    modifier: Modifier = Modifier,
    photoHeight: Dp = 160.dp,
    cornerRadius: Dp = 20.dp,
    drawableResId: Int = R.drawable.welcome_kitchen_main,
) {
    val shape = RoundedCornerShape(cornerRadius)
    Surface(
        modifier = modifier.fillMaxWidth(),
        shape = shape,
        color = MMColors.EvidenceCard,
        border = androidx.compose.foundation.BorderStroke(1.dp, MMColors.CardStroke),
    ) {
        Column {
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = 14.dp)
                    .padding(top = 12.dp, bottom = 10.dp),
                verticalAlignment = Alignment.CenterVertically,
            ) {
                Text(
                    "MOVE-IN PROOF",
                    fontSize = 10.sp,
                    fontWeight = FontWeight.SemiBold,
                    letterSpacing = 0.7.sp,
                    color = MMColors.TextMuted,
                    maxLines = 1,
                    modifier = Modifier.weight(1f, fill = false),
                )
                Text(
                    "SAVED",
                    fontSize = 11.sp,
                    fontWeight = FontWeight.SemiBold,
                    color = MMColors.Primary.copy(alpha = 0.95f),
                    modifier = Modifier
                        .background(MMColors.Primary.copy(alpha = 0.16f), RoundedCornerShape(50))
                        .border(0.85.dp, MMColors.Primary.copy(alpha = 0.48f), RoundedCornerShape(50))
                        .padding(horizontal = 9.dp, vertical = 5.dp),
                )
            }

            Box(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = 14.dp)
                    .height(photoHeight),
            ) {
                Box(
                    modifier = Modifier
                        .fillMaxSize()
                        .clip(RoundedCornerShape(14.dp))
                        .background(MMColors.FieldFill),
                ) {
                    Image(
                        painter = painterResource(drawableResId),
                        contentDescription = null,
                        contentScale = ContentScale.Crop,
                        modifier = Modifier.fillMaxSize(),
                    )
                    Box(
                        modifier = Modifier
                            .fillMaxWidth()
                            .height(56.dp)
                            .align(Alignment.BottomCenter)
                            .background(
                                androidx.compose.ui.graphics.Brush.verticalGradient(
                                    listOf(Color.Transparent, Color.Black.copy(alpha = 0.42f)),
                                ),
                            ),
                    )
                }
                if (tagsVisible) {
                    Row(
                        modifier = Modifier
                            .align(Alignment.BottomStart)
                            .padding(start = 12.dp, bottom = 12.dp, end = 8.dp),
                        horizontalArrangement = Arrangement.spacedBy(6.dp),
                    ) {
                        welcomeIssueTags.forEachIndexed { index, tag ->
                            val tagScale by animateFloatAsState(
                                if (tagsVisible) 1f else 0.92f,
                                MMMotion.welcomeTagEnterSpec(reduceMotion),
                                label = "welcomeTag$index",
                            )
                            Text(
                                tag,
                                fontSize = 10.sp,
                                fontWeight = FontWeight.SemiBold,
                                color = Color.White.copy(alpha = 0.95f),
                                maxLines = 1,
                                modifier = Modifier
                                    .scale(tagScale)
                                    .background(Color.Black.copy(alpha = 0.72f), RoundedCornerShape(8.dp))
                                    .padding(horizontal = 8.dp, vertical = 4.dp),
                            )
                        }
                    }
                }
            }

            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = 14.dp)
                    .padding(bottom = 12.dp),
            ) {
                Box(
                    modifier = Modifier
                        .fillMaxWidth()
                        .height(1.dp)
                        .padding(top = 12.dp)
                        .background(MMColors.CardStroke.copy(alpha = 0.65f)),
                )
                Row(
                    modifier = Modifier.padding(top = 10.dp),
                    verticalAlignment = Alignment.CenterVertically,
                ) {
                    Icon(
                        Icons.Filled.CheckCircle,
                        contentDescription = null,
                        tint = MMColors.Primary.copy(alpha = 0.95f),
                        modifier = Modifier.size(16.dp),
                    )
                    Spacer(Modifier.width(6.dp))
                    Text(
                        "Kitchen documented",
                        fontSize = 15.sp,
                        fontWeight = FontWeight.SemiBold,
                        color = MMColors.TextPrimary,
                        maxLines = 1,
                    )
                }
                Text(
                    "12 photos · Verified Apr 14",
                    fontSize = 12.sp,
                    fontWeight = FontWeight.Medium,
                    color = MMColors.TextMuted,
                    modifier = Modifier.padding(top = 4.dp),
                )
            }
        }
    }
}
